bentoml/_internal/yatai_rest_api_client/schemas.py

```diff
@@ -0,0 +1,247 @@
+class UserSchema:
+    name: str
+    email: str
+    first_name: str
+    last_name: str
+
+    def get_name(self) -> str:
+        if not self.first_name and not self.last_name:
+            return self.name
+        return f'{self.first_name} {self.last_name}'.strip()
+
+
+@attr.define
+class OrganizationSchema(ResourceSchema):
+    description: str
+
+
+@attr.define
+class OrganizationListSchema(BaseListSchema):
+    items: List[OrganizationSchema]
+
+
+@attr.define
+class ClusterSchema(ResourceSchema):
+    description: str
+
+
+@attr.define
+class CreateBentoRepositorySchema:
+    name: str
+    description: str
+
+
+class BentoImageBuildStatus(Enum):
+    PENDING = 'pending'
+    BUILDING = 'building'
+    SUCCESS = 'success'
+    FAILED = 'failed'
+
+
+class BentoUploadStatus(Enum):
+    PENDING = 'pending'
+    BUILDING = 'uploading'
+    SUCCESS = 'success'
+    FAILED = 'failed'
+
+
+@attr.define
+class BentoApiSchema:
+    route: str
+    doc: str
+    input: str
+    output: str
+
+
+@attr.define
+class BentoManifestSchema:
+    service: str
+    bentoml_version: str
+    apis: Dict[str, BentoApiSchema] = attr.field(factory=dict)
+    models: List[str] = attr.field(factory=list)
+
+
+@attr.define
+class BentoSchema(ResourceSchema):
+    description: str
+    version: str
+    image_build_status: BentoImageBuildStatus
+    upload_status: BentoUploadStatus
+    upload_finished_reason: str
+    presigned_upload_url: str
+    presigned_download_url: str
+    manifest: BentoManifestSchema
+
+    upload_started_at: Optional[datetime] = attr.field(default=None)
+    upload_finished_at: Optional[datetime] = attr.field(default=None)
+    build_at: datetime = attr.field(factory=datetime.now)
+
+
+@attr.define
+class BentoRepositorySchema(ResourceSchema):
+    description: str
+    latest_bento: Optional[BentoSchema]
+
+
+@attr.define
+class CreateBentoSchema:
+    description: str
+    version: str
+    manifest: BentoManifestSchema
+    build_at: datetime = attr.field(factory=datetime.now)
+    labels: List[LabelItemSchema] = attr.field(factory=list)
+
+
+@attr.define
+class FinishUploadBentoSchema:
+    status: Optional[BentoUploadStatus]
+    reason: Optional[str]
+
+@attr.define
+class CreateModelRepositorySchema:
+    name: str
+    description: str
+
+
+class ModelImageBuildStatus(Enum):
+    PENDING = 'pending'
+    BUILDING = 'building'
+    SUCCESS = 'success'
+    FAILED = 'failed'
+
+
+class ModelUploadStatus(Enum):
+    PENDING = 'pending'
+    BUILDING = 'uploading'
+    SUCCESS = 'success'
+    FAILED = 'failed'
+
+
+@attr.define
+class ModelManifestSchema:
+    module: str
+    api_version: str
+    bentoml_version: str
+    metadata: Dict[str, Any] = attr.field(factory=dict)
+    context: Dict[str, Any] = attr.field(factory=dict)
+    options: Dict[str, Any] = attr.field(factory=dict)
+
+
+@attr.define
+class ModelSchema(ResourceSchema):
+    description: str
+    version: str
+    image_build_status: ModelImageBuildStatus
```

Comment:
should model image build status be exposed here? My understanding is that it is a platform internal detail and not suppose to be user facing